src/PerfView/OtherSources/Linux/LinuxPerfScriptEventParser.cs

```diff
@@ -262,7 +263,22 @@ internal static string GetFileNameWithoutExtension(string path)
             var end = path.LastIndexOf('.');
             if (end < start)
                 end = path.Length;
-            return path.Substring(start, end - start);
+            var name = path.Substring(start, end - start);
+
+            if (stripNiSuffix)
+            {
+                var suffixStart = name.IndexOf(NISuffix);
+                var suffixEnd = suffixStart + NISuffix.Length - 1; // Leave the trailing '.' from ".ni."
+                start = 0;
+                end = name.Length;
```

Comment:
The assignments of start and end are dead at this point.  You should remove them.